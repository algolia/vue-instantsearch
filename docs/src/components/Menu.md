---
title: Menu
mainTitle: Components
layout: main.pug
category: Components
withHeadings: true
navWeight: 6
editable: true
githubSource: docs/src/components/Menu.md
---

Create a menu based on a facet. A menu displays a list of facet values and let the user selects only one value at a time. Make sure that the attribute you used is in `attributesForFaceting`.

<a class="btn btn-static-theme" href="stories/?selectedKind=Menu">ðŸ•¹ try out live</a>

## Usage

```html
<ais-menu :attribute="category"></ais-menu>
```

## Props

Name | Type | Default | Description | Required
---|---|---|---|---
attribute | string | - | The attribute | Yes
limit | number | 10 | Number of items to show | -
showMoreLimit | number | 20 | Number of items to show when the user clicked on "show more" | -
showMore | boolean | `false` | Whether or not to have the option to load more values | -
sortBy | string[] or function | `['name:asc', 'count:desc']` | Array or function to sort the results by | -
transformItems | `(items: object[]) => object[]` | `x => x` | Function which receives the items, which will be called before displaying them. Should return a new array with the same shape as the original array. Useful for mapping over the items to transform, remove or reorder them | -
classNames | Object | | Override class names | no

## CSS classes

Here's a list of CSS classes exposed by this widget. To better understand the underlying DOM structure, have a look at the generated DOM in your browser.

Note that you can pass the prop `class-names`, with an object of class names and their replacement to override this.

Class name | Description
---|---
`.ais-Menu` | the root div of the widget
`.ais-Menu--noRefinement` | the root div of the widget with no refinement
`.ais-Menu-list` | the list of all menu items
`.ais-Menu-item` | the menu list item
`.ais-Menu-item--selected` | the selected menu list item
`.ais-Menu-link` | the clickable menu element
`.ais-Menu-label` | the label of each item
`.ais-Menu-count` | the count of values for each item
`.ais-Menu-showMore` | the button used to display more categories
`.ais-Menu-showMore--disabled` | the disabled button used to display more categories
