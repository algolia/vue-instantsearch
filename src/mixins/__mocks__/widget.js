let state = {};

// we need to have state given by `component` before it is mounted, otherwise
// we can't render it in most cases (items, hits, etc. are used in the template)
// so we share a (mock) global state during a whole test.
//
// (a mock is imported once per test file, so the state is isolated between tests)
//
// This allows us to import this `__setState` function and call it in the test
// to give the necessary data before mounting.
export function __setState(newState) {
  state = newState;
}

export const createWidgetMixin = () => ({
  data() {
    return {
      state,
    };
  },
});
