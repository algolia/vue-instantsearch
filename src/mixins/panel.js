import Vue from 'vue';

export const PANEL_EMITTER_NAMESPACE = 'instantSearchPanelEmitter';
export const PANEL_CHANGE_EVENT = 'PANEL_CHANGE_EVENT';

export const createPanelProviderMixin = () => ({
  props: {
    emitter: {
      type: Object,
      required: false,
      default() {
        return new Vue({
          name: 'PanelProvider',
        });
      },
    },
  },
  provide() {
    return {
      [PANEL_EMITTER_NAMESPACE]: this.emitter,
    };
  },
  data() {
    return {
      canRefine: true,
    };
  },
  created() {
    this.emitter.$on(PANEL_CHANGE_EVENT, value => {
      this.updateCanRefine(value);
    });
  },
  beforeDestroy() {
    this.emitter.$destroy();
  },
  methods: {
    updateCanRefine(value) {
      this.canRefine = value;
    },
  },
});

export const createPanelConsumerMixin = ({ mapStateToCanRefine }) => ({
  inject: {
    emitter: {
      from: PANEL_EMITTER_NAMESPACE,
      default() {
        return {
          $emit: () => {},
        };
      },
    },
  },
  data() {
    return {
      state: null,
      hasAlreadyEmitted: false,
    };
  },
  watch: {
    state: {
      immediate: true,
      handler(nextState, previousState) {
        if (!nextState) {
          return;
        }

        const previousCanRefine = mapStateToCanRefine(previousState || {});
        const nextCanRefine = mapStateToCanRefine(nextState);

        if (!this.hasAlreadyEmitted || previousCanRefine !== nextCanRefine) {
          this.emitter.$emit(PANEL_CHANGE_EVENT, nextCanRefine);

          this.hasAlreadyEmitted = true;
        }
      },
    },
  },
});
