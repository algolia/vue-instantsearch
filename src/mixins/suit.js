import suit from '../util/suit';

export const createSuitMixin = ({ name }) => ({
  props: {
    classNames: {
      type: Object,
      default: undefined,
    },
  },
  methods: {
    suit(element, modifier) {
      const className = suit(name, element, modifier);
      const userClassName = this.classNames && this.classNames[className];
      if (userClassName) {
        return [className, userClassName].join(' ');
      }
      return className;
    },
  },
});
