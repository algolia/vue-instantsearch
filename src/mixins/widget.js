const defaultShouldUpdate = (previous, next) => previous !== next;

export const createWidgetMixin = ({ connect }) => ({
  inject: {
    instantSearchInstance: {
      name: 'instantSearchInstance',
      default() {
        const tag = this.$options._componentTag;
        throw new TypeError(
          `It looks like you forgot to wrap your Algolia search component
          "<${tag}>" inside of an "<ais-index>" component.`
        );
      },
    },
  },
  data() {
    return {
      state: null,
    };
  },
  computed: {
    widgetParams() {
      const { widgetParams } = this.$options;

      return Object.keys(widgetParams).reduce(
        (acc, name) => ({
          ...acc,
          [name]: widgetParams[name].getValue(this),
        }),
        {}
      );
    },
  },
  watch: {
    widgetParams(next, previous) {
      const { widgetParams } = this.$options;

      const shouldUpdateWidget = Object.keys(widgetParams).some(name => {
        const { shouldUpdate = defaultShouldUpdate } = widgetParams[name];

        return shouldUpdate(previous[name], next[name]);
      });

      if (shouldUpdateWidget) {
        this.instantSearchInstance.removeWidget(this.widget);
        this.widget = this.widgetFactory(next);
        this.instantSearchInstance.addWidget(this.widget);
      }
    },
  },
  created() {
    this.widgetFactory = connect(this.updateData, () => {});
    this.widget = this.widgetFactory(this.widgetParams);
    this.instantSearchInstance.addWidget(this.widget);
  },
  beforeDestroy() {
    this.instantSearchInstance.removeWidget(this.widget);
  },
  methods: {
    updateData(state = {}) {
      this.state = state;
    },
  },
});
