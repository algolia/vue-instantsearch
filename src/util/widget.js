export const addWidget = (connector, parent, widgetParams, render) => {
  const factory = connector(render);
  const widget = factory(widgetParams);
  parent.addWidgets([widget]);

  return widget;
};

export const forceRender = (widget, parent, instantSearchInstance) => {
  if (
    instantSearchInstance.__initialSearchResults &&
    !instantSearchInstance.started
  ) {
    if (typeof instantSearchInstance.__forceRender !== 'function') {
      throw new Error(
        'You are using server side rendering with <ais-instant-search> instead of <ais-instant-search-ssr>.'
      );
    }
    instantSearchInstance.__forceRender(widget, parent);
  }
};

export const removeWidget = (widget, parent) => {
  if (widget) {
    parent.removeWidgets([widget]);
  }
};
