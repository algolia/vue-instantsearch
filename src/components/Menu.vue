<template>
  <div
    v-if="state"
    :class="[suit(), !state.canRefine && suit('', 'noRefinement')]"
  >
    <slot
      :items="state.items"
      :can-refine="state.canRefine"
      :can-toggle-show-more="state.canToggleShowMore"
      :is-showing-more="state.isShowingMore"
      :refine="state.refine"
      :createURL="state.createURL"
      :toggle-show-more="state.toggleShowMore"
    >
      <ul :class="suit('list')">
        <li
          v-for="item in state.items"
          :key="item.value"
          :class="[suit('item'), item.isRefined && suit('item', 'selected')]"
        >
          <a
            :href="state.createURL(item.value)"
            :class="suit('link')"
            @click.prevent="state.refine(item.value)"
          >
            <span :class="suit('label')">{{ item.label }}</span>
            <span :class="suit('count')">{{ item.count }}</span>
          </a>
        </li>
      </ul>

      <button
        v-if="showShowMoreButton"
        :class="[suit('showMore'), !state.canToggleShowMore && suit('showMore', 'disabled')]"
        :disabled="!state.canToggleShowMore"
        @click.prevent="state.toggleShowMore()"
      >
        <slot
          name="showMoreLabel"
          :is-showing-more="state.isShowingMore"
        >{{ state.isShowingMore ? 'Show less' : 'Show more' }}</slot>
      </button>
    </slot>
  </div>
</template>

<script>
import { connectMenu } from 'instantsearch.js/es/connectors';
import { createPanelConsumerMixin } from '../mixins/panel';
import { createWidgetMixin } from '../mixins/widget';
import { createSuitMixin } from '../mixins/suit';

export default {
  name: 'AisMenu',
  mixins: [
    createSuitMixin({ name: 'Menu' }),
    createWidgetMixin({ connector: connectMenu }),
    createPanelConsumerMixin({
      mapStateToCanRefine: state => state.canRefine,
    }),
  ],
  props: {
    attribute: {
      type: String,
      required: true,
    },
    // TODO: implement searchable in connector
    // searchable: {
    //   type: Boolean,
    //   default: false,
    // },
    limit: {
      type: Number,
      default: 10,
    },
    showMoreLimit: {
      type: Number,
      default: 20,
    },
    showMore: {
      type: Boolean,
      default: false,
    },
    sortBy: {
      type: [Array, Function],
      default() {
        return ['count:desc', 'name:asc'];
      },
    },
    transformItems: {
      type: Function,
      default(items) {
        return items;
      },
    },
  },
  computed: {
    widgetParams() {
      return {
        attribute: this.attribute,
        limit: this.limit,
        showMore: this.showMore,
        showMoreLimit: this.showMoreLimit,
        sortBy: this.sortBy,
        transformItems: this.transformItems,
      };
    },
    showShowMoreButton() {
      return this.state.canRefine && this.showMore;
    },
  },
};
</script>
