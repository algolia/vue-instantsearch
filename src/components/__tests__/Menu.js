import { mount } from '@vue/test-utils';
import { __setState } from '../../mixins/widget';
import Menu from '../Menu.vue';

jest.mock('../../mixins/widget');
jest.mock('../../mixins/panel');

const apple = {
  value: 'Apple',
  label: 'Apple',
  count: 50,
  isRefined: false,
};

const samsung = {
  value: 'Samsung',
  label: 'Samsung',
  count: 25,
  isRefined: false,
};

const microsoft = {
  value: 'Microsoft',
  label: 'Microsoft',
  count: 12,
  isRefined: false,
};

const defaultState = {
  items: [apple, samsung, microsoft],
  canRefine: true,
  canToggleShowMore: false,
  isShowingMore: false,
  refine: () => {},
  createURL: () => {},
  toggleShowMore: () => {},
};

const defaultProps = {
  attribute: 'brand',
};

it('accepts an attribute prop', () => {
  __setState({
    ...defaultState,
  });

  const props = {
    ...defaultProps,
  };

  const wrapper = mount(Menu, {
    propsData: props,
  });

  expect(wrapper.vm.widgetParams.attribute).toBe('brand');
});

it('accepts a limit prop', () => {
  __setState({
    ...defaultState,
  });

  const props = {
    ...defaultProps,
    limit: 5,
  };

  const wrapper = mount(Menu, {
    propsData: props,
  });

  expect(wrapper.vm.widgetParams.limit).toBe(5);
});

it('accepts a showMoreLimit prop', () => {
  __setState({
    ...defaultState,
  });

  const props = {
    ...defaultProps,
    showMoreLimit: 10,
  };

  const wrapper = mount(Menu, {
    propsData: props,
  });

  expect(wrapper.vm.widgetParams.showMoreLimit).toBe(10);
});

it('accepts a sortBy prop', () => {
  __setState({
    ...defaultState,
  });

  const props = {
    ...defaultProps,
    sortBy: ['name:desc'],
  };

  const wrapper = mount(Menu, {
    propsData: props,
  });

  expect(wrapper.vm.widgetParams.sortBy).toEqual(['name:desc']);
});

it('accepts a transformItems prop', () => {
  __setState({
    ...defaultState,
  });

  const transformItems = () => {};

  const props = {
    ...defaultProps,
    transformItems,
  };

  const wrapper = mount(Menu, {
    propsData: props,
  });

  expect(wrapper.vm.widgetParams.transformItems).toBe(transformItems);
});

describe('default render', () => {
  it('renders correctly', () => {
    __setState({ ...defaultState });

    const wrapper = mount(Menu, {
      propsData: defaultProps,
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('renders correctly without refinement', () => {
    __setState({
      ...defaultState,
      items: [],
      canRefine: false,
    });

    const wrapper = mount(Menu, {
      propsData: defaultProps,
    });

    expect(wrapper.findAll('.ais-Menu--noRefinement')).toHaveLength(1);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('renders correctly with a selected item', () => {
    __setState({
      ...defaultState,
      items: [
        apple,
        {
          ...samsung,
          isRefined: true,
        },
        microsoft,
      ],
    });

    const wrapper = mount(Menu, {
      propsData: defaultProps,
    });

    expect(wrapper.findAll('.ais-Menu-item--selected')).toHaveLength(1);
    expect(wrapper.find('.ais-Menu-item--selected').text()).toBe('Samsung 25');
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('renders correctly with a URL for the href', () => {
    __setState({
      ...defaultState,
      createURL: value => `/brand/${value}`,
    });

    const wrapper = mount(Menu, {
      propsData: defaultProps,
    });

    expect(wrapper.find('.ais-Menu-link').attributes().href).toBe(
      '/brand/Apple'
    );
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('renders correctly with show more button', () => {
    __setState({ ...defaultState });

    const wrapper = mount(Menu, {
      propsData: {
        ...defaultProps,
        showMore: true,
      },
    });

    expect(wrapper.findAll('.ais-Menu-showMore')).toHaveLength(1);
    expect(wrapper.find('.ais-Menu-showMore').text()).toBe('Show more');
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('renders correctly with show more button toggled', () => {
    __setState({
      ...defaultState,
      isShowingMore: true,
    });

    const wrapper = mount(Menu, {
      propsData: {
        ...defaultProps,
        showMore: true,
      },
    });

    expect(wrapper.find('.ais-Menu-showMore').text()).toBe('Show less');
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('renders correctly with a disabled show more button', () => {
    __setState({
      ...defaultState,
      canToggleShowMore: false,
    });

    const wrapper = mount(Menu, {
      propsData: {
        ...defaultProps,
        showMore: true,
      },
    });

    const showMoreWrapper = wrapper.find('.ais-Menu-showMore');

    expect(wrapper.findAll('.ais-Menu-showMore')).toHaveLength(1);
    expect(showMoreWrapper.classes()).toContain('ais-Menu-showMore--disabled');
    expect(showMoreWrapper.attributes().disabled).toBe('disabled');
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('renders correctly without a show more button (canRefine)', () => {
    __setState({
      ...defaultState,
      items: [],
      canRefine: false,
    });

    const wrapper = mount(Menu, {
      propsData: {
        ...defaultProps,
        showMore: true,
      },
    });

    expect(wrapper.findAll('.ais-Menu-showMore')).toHaveLength(0);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('renders correctly without a show more button (showMore)', () => {
    __setState({ ...defaultState });

    const wrapper = mount(Menu, {
      propsData: {
        ...defaultProps,
        showMore: false,
      },
    });

    expect(wrapper.findAll('.ais-Menu-showMore')).toHaveLength(0);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('calls refine on link click', () => {
    const refine = jest.fn();

    __setState({
      ...defaultState,
      refine,
    });

    const wrapper = mount(Menu, {
      propsData: defaultProps,
    });

    wrapper.find('.ais-Menu-link').trigger('click');

    expect(refine).toHaveBeenCalledTimes(1);
    expect(refine).toHaveBeenCalledWith('Apple');
  });

  it('calls toggleShowMore on button click', () => {
    const toggleShowMore = jest.fn();

    __setState({
      ...defaultState,
      canToggleShowMore: true,
      toggleShowMore,
    });

    const wrapper = mount(Menu, {
      propsData: {
        ...defaultProps,
        showMore: true,
      },
    });

    wrapper.find('.ais-Menu-showMore').trigger('click');

    expect(toggleShowMore).toHaveBeenCalledTimes(1);
  });

  it('calls the Panel mixin with `canRefine`', () => {
    __setState({ ...defaultState });

    const wrapper = mount(Menu, {
      propsData: defaultProps,
    });

    const mapStateToCanRefine = () =>
      wrapper.vm.mapStateToCanRefine(wrapper.vm.state);

    expect(mapStateToCanRefine()).toBe(true);

    wrapper.setData({
      state: {
        canRefine: false,
      },
    });

    expect(mapStateToCanRefine()).toBe(false);
  });
});

describe('custom default render', () => {
  const defaultScopedSlot = `
    <div
      slot-scope="state"
      :class="[!state.canRefine && 'no-refinement']"
    >
      <ol>
        <li
          v-for="item in state.items"
          :key="item.value"
        >
          <a
            :href="state.createURL(item.value)"
            @click.prevent="state.refine(item.value)"
          >
            {{item.label}} - {{item.count}}
          </a>
        </li>
      </ol>
      <button
        :disabled="!state.canToggleShowMore"
        @click.prevent="state.toggleShowMore"
      >
        {{ state.isShowingMore ? 'Show less' : 'Show more' }}
      </button>
    </div>
  `;

  it('renders correctly', () => {
    __setState({ ...defaultState });

    const wrapper = mount(Menu, {
      propsData: defaultProps,
      scopedSlots: {
        default: defaultScopedSlot,
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('renders correctly without refinement', () => {
    __setState({
      ...defaultState,
      items: [],
      canRefine: false,
    });

    const wrapper = mount(Menu, {
      propsData: defaultProps,
      scopedSlots: {
        default: defaultScopedSlot,
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('renders correctly with an URL for the href', () => {
    __setState({
      ...defaultState,
      createURL: value => `/brand/${value}`,
    });

    const wrapper = mount(Menu, {
      propsData: defaultProps,
      scopedSlots: {
        default: defaultScopedSlot,
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('renders correctly with a show more button toggled', () => {
    __setState({
      ...defaultState,
      isShowingMore: true,
    });

    const wrapper = mount(Menu, {
      propsData: defaultProps,
      scopedSlots: {
        default: defaultScopedSlot,
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('renders correctly with a disabled show more button', () => {
    __setState({
      ...defaultState,
      canToggleShowMore: false,
    });

    const wrapper = mount(Menu, {
      propsData: defaultProps,
      scopedSlots: {
        default: defaultScopedSlot,
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('calls refine on link click', () => {
    const refine = jest.fn();

    __setState({
      ...defaultState,
      refine,
    });

    const wrapper = mount(Menu, {
      propsData: defaultProps,
      scopedSlots: {
        default: defaultScopedSlot,
      },
    });

    wrapper.find('a').trigger('click');

    expect(refine).toHaveBeenCalledTimes(1);
    expect(refine).toHaveBeenCalledWith('Apple');
  });

  it('calls toggleShowMore on button click', () => {
    const toggleShowMore = jest.fn();

    __setState({
      ...defaultState,
      canToggleShowMore: true,
      toggleShowMore,
    });

    const wrapper = mount(Menu, {
      propsData: {
        ...defaultProps,
      },
      scopedSlots: {
        default: defaultScopedSlot,
      },
    });

    wrapper.find('button').trigger('click');

    expect(toggleShowMore).toHaveBeenCalledTimes(1);
  });
});

describe('custom showMoreLabel render', () => {
  const showMoreLabelScopedSlot = `
    <span slot-scope="{ isShowingMore }">
      {{ isShowingMore ? 'Voir moins' : 'Voir plus' }}
    </span>
  `;

  it('renders correctly with a custom show more label', () => {
    __setState({ ...defaultState });

    const wrapper = mount(Menu, {
      propsData: {
        ...defaultProps,
        showMore: true,
      },
      scopedSlots: {
        showMoreLabel: showMoreLabelScopedSlot,
      },
    });

    expect(wrapper.find('.ais-Menu-showMore').text()).toBe('Voir plus');
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('renders correctly with a custom show more label toggled', () => {
    __setState({
      ...defaultState,
      isShowingMore: true,
    });

    const wrapper = mount(Menu, {
      propsData: {
        ...defaultProps,
        showMore: true,
      },
      scopedSlots: {
        showMoreLabel: showMoreLabelScopedSlot,
      },
    });

    expect(wrapper.find('.ais-Menu-showMore').text()).toBe('Voir moins');
    expect(wrapper.html()).toMatchSnapshot();
  });
});
