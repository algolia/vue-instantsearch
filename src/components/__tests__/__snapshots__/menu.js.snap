// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `

<div class="ais-Menu">
  <ul class="ais-Menu-list">
    <li class="ais-Menu-item--active">
      <a class="ais-Menu-link">
        <span class="ais-Menu-label">
          foo
        </span>
        <span class="ais-Menu-count">
          2
        </span>
      </a>
    </li>
    <li class="ais-Menu-item">
      <a class="ais-Menu-link">
        <span class="ais-Menu-label">
          bar
        </span>
        <span class="ais-Menu-count">
          3
        </span>
      </a>
    </li>
    <li class="ais-Menu-item">
      <a class="ais-Menu-link">
        <span class="ais-Menu-label">
          foobar
        </span>
        <span class="ais-Menu-count">
          4
        </span>
      </a>
    </li>
  </ul>
</div>

`;

exports[`renders correctly with showMore 1`] = `

<div class="ais-Menu">
  <ul class="ais-Menu-list">
    <li class="ais-Menu-item--active">
      <a class="ais-Menu-link">
        <span class="ais-Menu-label">
          foo
        </span>
        <span class="ais-Menu-count">
          2
        </span>
      </a>
    </li>
    <li class="ais-Menu-item">
      <a class="ais-Menu-link">
        <span class="ais-Menu-label">
          bar
        </span>
        <span class="ais-Menu-count">
          3
        </span>
      </a>
    </li>
    <li class="ais-Menu-item">
      <a class="ais-Menu-link">
        <span class="ais-Menu-label">
          foobar
        </span>
        <span class="ais-Menu-count">
          4
        </span>
      </a>
    </li>
  </ul>
  <button class="ais-Menu-showMore">
    Show more
  </button>
</div>

`;
